types/tx/middleware.go

```diff
@@ -0,0 +1,33 @@
+package tx
+
+import (
+	context "context"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// RequestSimulateTx is the request type for the TxHandler.RequestSimulateTx
+// method.
+type RequestSimulateTx struct {
+	TxBytes []byte
+}
+
+// ResponseSimulateTx is the response type for the TxHandler.RequestSimulateTx
+// method.
+type ResponseSimulateTx struct {
+	GasInfo sdk.GasInfo
+	Result  *sdk.Result
+}
+
+// TxHandler defines the baseapp's CheckTx, DeliverTx and Simulate respective
+// handlers. It is designed as a middleware stack.
+type TxHandler interface {
```

Comment:
linter says `tx.TxHandler`. If there's no objection, I'm going to:
```suggestion
type Handler interface {
```

Reply:
+1.   we are in the `tx` package, so `Handler` is enough.